src/merge.py

```diff
@@ -23,16 +25,19 @@ def merge(x: list[int], y: list[int]) -> list[int]:
         if x[i] <= y[j]: 
             z.append(x[i])
             i += 1
-        if y[j] <= x[i]: 
+        elif y[j] <= x[i]: 
```

Comment:
You could also use `else` here. If the two are equal, it doesn't matter which you use (and you will use `x[i]` if you have the test above), so an `else` happens when `y[j]` is the smaller.